File: examples/documentation/payload-comparison/content/posts.go
package content

import (
	"github.com/riducms/ridu"
	"github.com/riducms/ridu/field"
)

func signedIn(ctx ridu.AccessContext) (ridu.AccessDecision, error) {
	// Actor is nil when no user is signed in.
	if ctx.Actor == nil {
		// Denying access is a decision; nil means no execution error.
		return ridu.Deny(), nil
	}
	return ridu.Allow(), nil
}

var Posts = ridu.Collection{
	Slug: "posts",
	Admin: ridu.CollectionAdmin{
		UseAsTitle:     "title",
		DefaultColumns: []string{"title", "author", "updatedAt"},
	},
	Fields: field.Fields{
		field.Text("title").Required().MaxLength(120),
		field.Text("slug").Required().Unique().Index(),
		field.Textarea("summary"),
		// Store the user's ID. Ownership permissions come later.
		field.Relationship("author", "users").Required(),
	},
	Access: ridu.CollectionAccess{
		Read: func(ridu.AccessContext) (ridu.AccessDecision, error) {
			return ridu.Allow(), nil
		},
		Create: signedIn,
		Update: signedIn,
		Delete: signedIn,
	},
}

File: examples/documentation/payload-comparison/payload/posts.ts
import type { Access, CollectionConfig } from 'payload';

const signedIn: Access = ({ req: { user } }) => Boolean(user);

export const Posts: CollectionConfig = {
	slug: 'posts',
	// Match Ridu's default: document locking is off.
	lockDocuments: false,
	admin: {
		useAsTitle: 'title',
		defaultColumns: ['title', 'author', 'updatedAt']
	},
	fields: [
		{
			name: 'title',
			type: 'text',
			required: true,
			maxLength: 120
		},
		{
			name: 'slug',
			type: 'text',
			required: true,
			unique: true,
			index: true
		},
		{ name: 'summary', type: 'textarea' },
		// Store the user's ID. Ownership permissions come later.
		{
			name: 'author',
			type: 'relationship',
			relationTo: 'users',
			required: true
		}
	],
	access: {
		read: () => true,
		create: signedIn,
		update: signedIn,
		delete: signedIn
	}
};
